guides/source/active_record_callbacks.md

```diff
@@ -554,6 +554,8 @@ WARNING. The code executed within `after_commit` or `after_rollback` callbacks i
 
 WARNING. Using both `after_create_commit` and `after_update_commit` with the same method name will only allow the last callback defined to take effect, as they both internally alias to `after_commit` which overrides previously defined callbacks with the same method name.
 
+WARNING. In the context of a single transaction, should you interact with multiple instances of the same model, `after_commit` or `after_rollback` callback behavior is designed to execute callbacks for only the first instance of the model. Subsequent instances of the same model within that transaction will _not_ trigger their own `after_commit` or `after_rollback` callback. This is a nuanced behavior that can impact the predictability and flow of callback execution, particularly in scenarios where multiple instances of the same model are undergoing changes in a single transaction.
```

Comment:
Can you provide a simple example of what this would look like or how you're doing this transaction?

Reply:
https://github.com/rails/rails/issues/39400 
It's given in this PR. 
Or. did you mean i should add an example to the docs?

Reply:
I was thinking an actual code example but since it's mentioned in that PR never mind.

Reply:
updated the msg to be more clear. I think multiple instances might be confusing